src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -2542,5 +2542,93 @@ public void CallingPartialValidationSucceeded_FoHeaderWithValidationStateAsParti
+
+            // 10 headers are presented.
+            ChainedHeader extendedChainTip = testContext.ExtendAChain(chainExtensionSize, initialChainTip);
+            List<BlockHeader> listOfAlternativeChainBlockHeaders = testContext.ChainedHeaderToList(extendedChainTip, chainExtensionSize);
+
+            // 10 blocks are downloaded.
+            ConnectNewHeadersResult connectNewHeadersResult = chainedHeaderTree.ConnectNewHeaders(peerOneId, listOfAlternativeChainBlockHeaders);
+            Assert.Equal(connectNewHeadersResult.DownloadFrom.Header, extendedChainTip.GetAncestor(initialChainSize + 1).Header); // h1
+            Assert.Equal(connectNewHeadersResult.DownloadTo.Header, extendedChainTip.Header); // h10
+            Assert.True(connectNewHeadersResult.HaveBlockDataAvailabilityStateOf(BlockDataAvailabilityState.BlockRequired));
+
+            foreach (ChainedHeader chainedHeader in connectNewHeadersResult.ToHashArray())
+            {
+                chainedHeaderTree.BlockDataDownloaded(chainedHeader, extendedChainTip.FindAncestorOrSelf(chainedHeader).Block);
+            }
+
+            // CT advances to 5.
+            chainedHeaderTree.ConsensusTipChanged(connectNewHeadersResult.DownloadTo.GetAncestor(5));
+
+            // Alternative chain with fork at 3 and tip at 12 is presented.
+            const int heightOfFork = 3;
+            const int chainBExtension = 9;
+            ChainedHeader forkedChainHeader = connectNewHeadersResult.DownloadTo.GetAncestor(heightOfFork);
+            ChainedHeader alternativeChainTip = testContext.ExtendAChain(chainBExtension, forkedChainHeader);
+            Assert.Equal(12, alternativeChainTip.Height);
+
+            // Block data for alternative chain is downloaded.
+            listOfAlternativeChainBlockHeaders = testContext.ChainedHeaderToList(alternativeChainTip, alternativeChainTip.Height - heightOfFork);
+            connectNewHeadersResult = chainedHeaderTree.ConnectNewHeaders(peerTwoId, listOfAlternativeChainBlockHeaders);
+
+            Assert.True(connectNewHeadersResult.HaveBlockDataAvailabilityStateOf(BlockDataAvailabilityState.BlockRequired));
+            Assert.Equal(connectNewHeadersResult.DownloadFrom.Header, alternativeChainTip.GetAncestor(heightOfFork + 1).Header);
+            Assert.Equal(connectNewHeadersResult.DownloadTo.Header, alternativeChainTip.Header);
+
+            foreach (ChainedHeader chainedHeader in connectNewHeadersResult.ToHashArray())
+            {
+                chainedHeaderTree.BlockDataDownloaded(chainedHeader, alternativeChainTip.FindAncestorOrSelf(chainedHeader).Block);
+            }
+ 
+            // UnconsumedBlocksDataBytes is equal to the sum of serialized sizes of 9b-12b + 6a-10a.
+            int serializedSizeOfChainA = 0;
+            int serializedSizeOfChainB = 0;
+
+            ChainedHeader chainedHeaderChainA = alternativeChainTip;
+            while (chainedHeaderChainA.Height >= 9)
```

Comment:
You have swapped the chains here, check the assignment description.

Reply:
Well spotted.  Fixed.